### PR TITLE
Mutation improver

### PR DESCRIPTION
**Pull request by andyMrtnzP**
## Description

Modified MutationObserver in `content.js` to filter DOM mutations before iterating over them.

## Browsers

Tested and working as expected on:

- Google Chrome (96.0.4664.45).
- Mozilla Firefox (94.0.2).

### CODE REVIEW

File: scripts/content.js [hunk 1]
```diff
@@ -139,13 +139,13 @@ const observer = new MutationObserver((mutations, instance) => {
...
 
   if (!isPreview) {
-    mutations.forEach((mutation) => {
-      mutation.addedNodes.forEach((node) => {
-        const valid = check(node);
-
-        if (valid && node.matches(selectors)) node.outerHTML = "";
-      });
-    });
+    const nodes = mutations.map((mutation) => [...mutation.addedNodes])
```
Comment (wanhose) [inline scripts/content.js:142]: I really like the "more functional" approach but it seems a less performant way to iterate these arrays. Previously we had only two iterations using two `for each` loops and now we have four: `map`, `flat`, `filter` and finally a `for of` loop.

File: scripts/content.js [hunk 2]
```diff
@@ -139,13 +139,13 @@ const observer = new MutationObserver((mutations, instance) => {
...
+                           .flat()
+                           .filter((node) => check(node));
+    
+    for (const node of nodes) {
```
Comment (wanhose) [inline scripts/content.js:146]: Benchmarks say that a `for of` loop is more performant than a `for each` loop in some browsers. How about changing these two `for each` by two `for of` loops?

**Review by wanhose — CHANGES_REQUESTED** [review]
Welcome to the community @andyMrtnzP, 

You're the first code contributor to the project. I really like your suggestion but I have some comments. Please take a look. 😁

File: scripts/content.js (hunk 1 above)
Comment (andyMrtnzP) [reply scripts/content.js:142]: Oh, I clearly see what you mean. So sorry. I'll revert those changes

File: scripts/content.js (hunk 2 above)
Comment (andyMrtnzP) [reply scripts/content.js:146]: Yup, will do it!

**Comment by andyMrtnzP** [comment]
Reverted last commit. I also added element selectors for some websites I encountered that the extension doesn't cover yet (Libreriaberkana, Verso, Medium, Tidal, Medium, Mercadolibre and others)

**Comment by wanhose** [comment]
Wooow! Amazing job @andyMrtnzP!

I will check these new selectors and then I will merge everything into our next release 🚀

**Comment by wanhose** [comment]
Hi again @andyMrtnzP!

Could you resolve conflicts and squash all your commits into just one?

**Comment by wanhose** [comment]
Hi again @andyMrtnzP!

Could you resolve conflicts, squash all your commits into just one and set the target branch to `5.4.4` branch?

**Comment by wanhose** [comment]
Hi again @andyMrtnzP!

I'm going to close the pull request because I have already copied your commits to the release branch. You don't have to do nothing else.

Thank you again for your contribution.

**Pull request closed without merging**